src/rcc.rs

```diff
@@ -37,12 +47,36 @@ pub struct Rcc {
 
 #[allow(unused)]
 const HSI: u32 = 8_000_000; // Hz
+#[allow(dead_code)] // Only used for ADC, but no configuration for ADC clock yet.
+const HSI14: u32 = 14_000_000; // Hz - ADC clock.
+// Can't do this anymore... #[cfg(any(feature = "stm32f042", feature = "stm32f072"))]
+const HSI48: u32 = 48_000_000; // Hz - (available on STM32F04x, STM32F07x and STM32F09x devices only)
+
+pub enum SysClkSource {
+    HSI = 0b00,
+    HSE = 0b01,
+    PLL = 0b10,
+    HSI48 = 0b11,
+}
+
+pub enum PllSource {
+    HSI_2 = 0b00,
+    HSI = 0b01,
+    HSE = 0b10,
+    HSI48 = 0b11,
+}
 
 #[allow(unused)]
 pub struct CFGR {
     hclk: Option<u32>,
     pclk: Option<u32>,
     sysclk: Option<u32>,
+    enable_hsi: Option<bool>,
+    enable_hsi14: Option<bool>,
+    // This does not work very well: #[cfg(any(feature = "stm32f042", feature = "stm32f072"))]
+    enable_hsi48: Option<bool>,
```

Comment:
Is there a reason for these (except `enable_hsi`) being an Option?

Reply:
After re-reading your question and the one below I think I have this fixed now?

Reply:
Yeah, looks fine now

Reply:
Awesome :)